### PR TITLE
feat: private route and authentication

### PR DESCRIPTION
**Pull request by shristabaruah**
- authentication login signup 
- toasters and private route

### CODE REVIEW

**Review by JowelTisso — APPROVED** [review]
Looks good

File: src/contexts/wishlist-context.js [hunk 1]
```diff
@@ -1,26 +1,29 @@
...
-        try {
-          const response = await axios.get("/api/user/wishlist", {
-            headers: { authorization: token },
-          });
-          setWishlist(response.data.wishlist);
-        } catch (e) {
-          console.error("error:", e.response);
+        if (token) {
+          try {
+            const response = await axios.get("/api/user/wishlist", {
```
Comment (t007rushi) [inline src/contexts/wishlist-context.js:17]: destructure here itself {data}

File: src/index.js [hunk 2]
```diff
@@ -7,20 +7,23 @@ import { makeServer } from "./server";
 import { FilterProvider } from "./contexts/filter-context";
 import { WishlistProvider } from "./contexts/wishlist-context";
 import { CartProvider } from "./contexts/cart-context";
+import { AuthProvider } from "./contexts/auth-context";
```
Comment (t007rushi) [inline src/index.js:11]: can barrel import using ```index``` file

File: src/reducer/auth-reducer.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+const authReducer = (state, action) => {
```
Comment (t007rushi) [inline src/reducer/auth-reducer.js:1]: same here make use of destructure makes more readable action => {type. payload}

**Review by t007rushi — APPROVED** [review]
Else Everything looks good to me

**Pull request merged**